tests/nlu/featurizers/test_lexical_syntactic_featurizer.py

```diff
@@ -1,197 +1,312 @@
+    def inner(config: Dict[Text, Any]):
+        return LexicalSyntacticFeaturizerGraphComponent.create(
+            config=config,
+            model_storage=default_model_storage,
+            execution_context=default_execution_context,
+            resource=resource_lexical_syntactic_featurizer,
+        )
+
+    return inner
 
 
 @pytest.mark.parametrize(
-    "sentence, expected_features",
+    "sentence,part_of_speech,feature_config,expected_features",
     [
+        # simple example 1
         (
             "hello goodbye hello",
+            None,
+            [["BOS", "upper"], ["BOS", "EOS", "prefix2", "digit"], ["EOS", "low"],],
             [
                 [0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 1.0],
                 [0.0, 1.0, 1.0, 1.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0],
                 [1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 0.0],
-                [1.0, 1.0, 2.0, 2.0, 1.0, 2.0, 1.0, 3.0, 1.0, 2.0, 1.0, 1.0, 2.0],
             ],
         ),
+        # simple example 2
         (
             "a 1",
+            None,
+            [["BOS", "upper"], ["BOS", "EOS", "prefix2", "digit"], ["EOS", "low"],],
             [
                 [0.0, 0.0, 0.0, 1.0, 1.0, 0.0, 1.0, 0.0, 0.0, 1.0, 1.0, 1.0],
                 [1.0, 1.0, 1.0, 0.0, 0.0, 1.0, 0.0, 1.0, 1.0, 0.0, 0.0, 0.0],
-                [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+            ],
+        ),
+        # larger window size
+        (
+            "hello 123 hello 123 hello",
+            None,
+            [["upper"], ["digit"], ["low"], ["digit"]],
+            [0.0, 0.0, 0.0, 0.0, 1.0, 0.0, 1.0],
+            # Note:
+            # 1. we just describe the features for first token here
+            # 2. "123".islower() == "123".isupper() == False, which is why we end
+            #     up with 7 features
+        ),
+        # with part of speech
+        (
+            "The sun is shining",
+            ["DET", "NOUN", "AUX", "VERB"],
+            [["pos", "pos2"]],
+            [
+                [0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0],
+                [0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0, 0.0],
+                [1.0, 0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0],
+                [0.0, 0.0, 0.0, 1.0, 0.0, 0.0, 0.0, 1.0],
             ],
         ),
     ],
 )
-def test_text_featurizer(sentence, expected_features):
-    featurizer = LexicalSyntacticFeaturizer(
-        {
-            "features": [
-                ["BOS", "upper"],
-                ["BOS", "EOS", "prefix2", "digit"],
-                ["EOS", "low"],
-            ]
-        }
+def test_feature_computation(
+    create_lexical_syntactic_featurizer: Callable[
+        [Dict[Text, Any]], LexicalSyntacticFeaturizerGraphComponent
+    ],
+    sentence: Text,
+    part_of_speech: Optional[List[Text]],
+    feature_config: List[List[Text]],
+    expected_features: List[List[int]],
```

Comment:
Since one of the test cases is just `List[int]`

```suggestion
    expected_features: List[Union[int, List[int]]],
```